api/controllers/user.js

```diff
@@ -319,12 +342,59 @@ function loginUser(req, res) {
         birthdate: moment(user.birthdate).format('YYYY-MM-DD'),
         authToken: tokenString
       };
-
       return res.status(200).json(response);
     }
   });
 }
 
+async function updateUserLocation(ip, user) {
+  if (!user.location || !user.location.country) {
+    const newLocation = await getLocation(ip);
+    if (newLocation && newLocation.country) {
+      user.location = newLocation;
+      try {
+        const dbUser = await user.save();
+        if (!dbUser) {
+          user.location = {};
+          console.log("Unable to find user on the DB")
+          return;
+        }
+      }
+      catch (err) {
+        console.log("Error saving user location", err)
+        user.location = {};
```

Comment:
The same as the comment before
https://github.com/cboard-org/cboard-api/pull/204/files#r927164259